### PR TITLE
[DBT-456] Add code to fetching warehouse version

### PR DESCRIPTION
**Pull request by tovganesh**
## Describe your changes
Fetch warehouse version when available, and send it along the instrumentation data.
Only supported for Spark 3 and above

## Internal Jira ticket number or external issue link
DBT-456 (internal)

## Testing procedure/screenshots(if appropriate):
For a valid dbt profile, run dbt debug, the logs should produce an instrumentation record similar to the following:

10:27:21.132604 [debug] [Thread-6  ]: Tracker adapter: Sending Event [{"data": {"event_type": "dbt_spark_livy_warehouse", "warehouse_version": "[{\"version()\": \"3.2.1 f38f605165f50b0d9e887e279c350728b839915d\"}]", "auth": "N/A", "connection_state": "N/A", "elapsed_time": "N/A", "incremental_strategy": "N/A", "model_name": "N/A", "model_type": "N/A", "permissions": "N/A", "profile_name": "N/A", "sql_type": "N/A", "id": "dff525d4-219b-46cd-b58a-f5ceee395564", "unique_host_hash": "279763963ad1afa5776991f84894d563", "unique_user_hash": "e8a5d78ea632435d797d3d58df650872", "unique_session_hash": "ec5d62227c01b174fb67e6caf1dd6fe9", "python_version": "3.9.12", "system": "Darwin", "machine": "arm64", "platform": "macOS-12.6-arm64-arm-64bit", "dbt_version": "1.1.2", "dbt_adapter": "spark_livy-1.1.5", "ml_runtime_edition": "", "ml_runtime_git_hash": "", "ml_runtime_kernel": "", "ml_runtime_editor": "", "ml_runtime_gbn": "", "ml_runtime_full_version": "", "ml_runtime_description": "", "ml_runtime_maintenance_version": "", "ml_runtime_metadata_version": ""}}]

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] I have formatted my added/modified code to follow pep-8 standards
- [X] I have checked suggestions from python linter to make sure code is of good quality.

### CODE REVIEW

File: dbt/adapters/spark_livy/livysession.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def _getLivySQL(self, sql):
         # code = "val sprk_sql = spark.sql(\"" + escapedSQL + "\")\nval sprk_res=sprk_sql.collect\n%json sprk_res"  # .format(escapedSQL)
 
         # TODO: since the above code is not changed to sending direct SQL to the livy backend, client side string escaping is probably not needed
-        code = sql
+        code = sql.replace("\n", " ").replace("\t", " ")
```
Comment (TapasSenapati) [inline dbt/adapters/spark_livy/livysession.py]: why is this change needed

**Review by sujitkp-blr — DISMISSED** [review]

File: dbt/adapters/spark_livy/livysession.py (hunk 1 above)
Comment (tovganesh) [reply dbt/adapters/spark_livy/livysession.py]: good catch. removed this.

**Review by TapasSenapati — APPROVED** [review]
